### ISSUE TITLE
[Feature request] Flag to prevent certain parameters from being typecast by nn.module.to()

### ISSUE THREAD

**Issue by Ben-Schneider-code** [pytorch/pytorch#138952]
### 🚀 The feature, motivation and pitch

 I work with a lot of models that are mostly bf16 but have a couple parameters that I need to have stay in fp32. The most common ones are loss scaling parameters used for fp32 loss calculations that require grad updates.

A flag similar to ```requires_grad``` but for preventing casting would be very handy.
I.e. something similar to:
```
self.p = nn.Parameter(t)
self.p.dynamic_dtype = False
```

### Alternatives

There are workarounds (i.e. saving and restoring the parameter) but they tend to be quite error prone, especially when working with training libraires that cast the nn.module. You always have to be on the lookout whenever you're loading or training that a cast doesn't slip by.
